File: prime.c
#include <stdio.h>

static int prime(int num)
{
    int i;
    for (i = 2; i <= num/2; i++)
        if (num % i == 0)
            return 0;
    return 1;
}

int main()
{
    int num = 0;
    int i;
    for (i = 2; i <= 100000; i++)
        if (prime(i))
            num++;
    printf("number of primes = %d\n", num);

    return 0;
}

File: prime.rb
#!/usr/bin/ruby

def prime(num)
  for i in 2..num/2 do
    if num % i == 0 then
      return false
    end
  end
  return true
end

def main()
  num = 0
  for i in 2..100000 do
    if prime(i) then
      num = num + 1
    end
  end
  print("number of primes = ", num, "\n")
end

main()
